src/main/java/org/opensearch/security/tools/democonfig/Installer.java

```diff
@@ -0,0 +1,377 @@
+        + ".opendistro-reports-*, .opensearch-notifications-*, .opensearch-notebooks, .opensearch-observability, .ql-datasources, "
+        + ".opendistro-asynchronous-search-response*, .replication-metadata-store, .opensearch-knn-models, .geospatial-ip2geo-data*";
+
+    static SecuritySettingsConfigurer securitySettingsConfigurer;
+    static CertificateGenerator certificateGenerator;
+
+    public static void main(String[] options) {
+        securitySettingsConfigurer = new SecuritySettingsConfigurer();
+        certificateGenerator = new CertificateGenerator();
+
+        printScriptHeaders();
+        readOptions(options);
+        gatherUserInputs();
+        initializeVariables();
+        printVariables();
+        securitySettingsConfigurer.configureSecuritySettings();
+        certificateGenerator.createDemoCertificates();
+        finishScriptExecution();
+    }
+
+    /**
+     * Prints deprecation warning and other headers for the script
+     */
+    static void printScriptHeaders() {
+        System.out.println("**************************************************************************");
+        System.out.println("** This tool will be deprecated in the next major release of OpenSearch **");
+        System.out.println("** https://github.com/opensearch-project/security/issues/1755           **");
+        System.out.println("**************************************************************************");
+        System.out.println("\n");
+        System.out.println("### OpenSearch Security Demo Installer");
+        System.out.println("### ** Warning: Do not use on production or public reachable systems **");
+    }
+
+    /**
+     * Reads the options passed to the script
+     * @param options an array of strings containing options passed to the script
+     */
+    static void readOptions(String[] options) {
+        // set script execution dir
+        SCRIPT_DIR = options[0];
+
+        for (int i = 1; i < options.length; i++) {
+            switch (options[i]) {
+                case "-y":
+                    assumeyes = true;
+                    break;
+                case "-i":
+                    initsecurity = true;
+                    break;
+                case "-c":
+                    cluster_mode = true;
+                    break;
+                case "-s":
+                    skip_updates = 0;
+                    break;
+                case "-t":
+                    environment = ExecutionEnvironment.test;
+                    break;
+                case "-h":
+                case "-?":
+                    showHelp();
+                    return;
+                default:
+                    System.out.println("Invalid option: " + options[i]);
+            }
+        }
+    }
+
+    /**
+     * Prints the help menu when -h option is passed
+     */
+    static void showHelp() {
+        System.out.println("install_demo_configuration.sh [-y] [-i] [-c]");
```

Comment:
As soon as the min version of Java supported is >= 15 we will start being able to use [text blocks](https://docs.oracle.com/en/java/javase/21/text-blocks/index.html).